ddtrace/appsec/_asm_request_context.py

```diff
@@ -128,7 +152,7 @@ def call_waf_callback(custom_data=None):
 
 def set_ip(ip):  # type: (Optional[str]) -> None
     if ip is not None:
-        set_value("waf_addresses", SPAN_DATA_NAMES.REQUEST_HTTP_IP, ip)
+        set_waf_address(SPAN_DATA_NAMES.REQUEST_HTTP_IP, ip, _ASM.get().span)
```

Comment:
Can you explain why the change requires passing the asm request context's span? Why can't this be accomplished when the `_ASM.get()` is called to get the asm context in `set_value()`?

Reply:
Because now, we set the span context at the same time for metrics and tags purpose.